elastic/elastic-agent
# Add Resource Detection processor to OTel mode

## What does this PR do?

This PR makes the [Resource Detection processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md) available to users running Elastic Agent in `otel` mode.

## Why is it important?

This allows users to enrich the collected telemetry with metadata of the infrastructure, e.g. `host.name`, `os.type`, `cloud.provider`, `cloud.region`, `k8s.cluster.name` and many more.

## Checklist


## Disruptive User Impact

No impact expected.

## How to test this PR locally

1. Build the agent locally (see [docs](https://github.com/elastic/elastic-agent/blob/main/README.md#packaging))
2. Unpack the .tar.gz archive in `build/distributions`
3. Run the built binary with `build/distributions/elastic-agent-*/elastic-agent otel --config internal/pkg/otel/testdata/all-components.yml
